sql/slave.cc

```diff
@@ -3794,13 +3796,26 @@ sql_delay_event(Log_event *ev, THD *thd, rpl_group_info *rgi)
                         sql_delay, (long)ev->when,
                         rli->mi->clock_diff_with_master,
                         (long)now, (ulonglong)sql_delay_end, (long)nap_time));
-
-    if (sql_delay_end > now)
+    /* if using debug_sync for sql_delay, only delay once per event */
+    if (DBUG_EVALUATE_IF("sql_delay_by_debug_sync", type == GTID_EVENT,
+                         sql_delay_end > now))
```

Comment:
Do you mean here "only delay once per event group" ?

In any case, thanks Brandon for taking this and fixing the random failures in the test suite.